src/devices/src/virtio/block/device.rs

```diff
@@ -126,6 +139,7 @@ impl Block {
         Ok(Block {
             disk_image_id: build_disk_image_id(&disk_image),
```

Comment:
We're not saving this, but rebuilding it from File. Which is correct, but it got me thinking how do we protect ourselves against getting a _wrong_ File on restore...

Would it make sense to both save the `disk_image_id` and also compute it from File on restore and sanity check them against each other?

Same for `disk_size` which is used to to compute other internals.

Reply:
That's a good idea, but I'm not sure how we can decide if this is an error or not.
For example, if the backing file gets bigger it should not be a problem.

Should we log a warning maybe?